### PR TITLE
Adding librdkafka

### PR DESCRIPTION
**Pull request by kwilcox**
Linux only ATM but can be compiled for MacOS and Win fairly easily?  https://github.com/edenhill/librdkafka#instructions

### CODE REVIEW

File: recipes/librdkafka/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
+export CPPFLAGS="$CPPFLAGS -I$PREFIX/include"
+export C_INCLUDE_PATH="$PREFIX/include"
+export LIBRARY_PATH="$PREFIX/lib"
+
+./configure --prefix=$PREFIX
+make -j 2
```
Comment (ocefpaf) [inline recipes/librdkafka/build.sh]: Use the `CPU_COUNT` variable instead of hard-coding the number of cores.

File: recipes/librdkafka/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: {{ build }}
+  skip: True  # [win]
+
+requirements:
+  build:
+    - zlib 1.2.*
+    - openssl
```
Comment (ocefpaf) [inline recipes/librdkafka/meta.yaml]: ping to `openssl 1.0.*`

File: recipes/librdkafka/build.sh (hunk 1 above)
Comment (ocefpaf) [inline recipes/librdkafka/build.sh]: Is there a `make check`?

Comment (kwilcox) [inline recipes/librdkafka/build.sh]: kk was just following https://github.com/conda-forge/gdal-feedstock/blob/master/recipe/build.sh#L82

File: recipes/librdkafka/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - libgsasl
+    - lz4
+  run:
+    - zlib 1.2.*
+    - openssl
+    - libgsasl
+    - lz4
+
```
Comment (ocefpaf) [inline recipes/librdkafka/meta.yaml]: If this is a library only package we can add an "does it exist" test here like:

```yaml
test:
  commands:
    - test -f ${PREFIX}/lib/libpackage.so  # [lnux]
    - test -f ${PREFIX}/lib/libpackage.dylib  # [osx]
    - test -f ${PREFIX}/lib/libpackage.a  # [not win]
```

**Comment by ocefpaf** [comment]
@kwilcox `CircleCI` is not building forked PRs. Not sure if that feature will come back as it was considered a bug in their eyes :cry:

File: recipes/librdkafka/build.sh (hunk 1 above)
Comment (ocefpaf) [inline recipes/librdkafka/build.sh]: That guy, listed at the top of the maintainers list, is a lazy bum that forgot to update that recipe :stuck_out_tongue_winking_eye:

**Comment by kwilcox** [comment]
@ocefpaf rdy for review again

**Comment by ocefpaf** [comment]
@kwilcox can you add a `skip` for osx too?

**Comment by ocefpaf** [comment]
Weird I refreshed the page and still so a failure on OS X. Now it is gone! I guess my connection was playing tricks with me.

**Pull request merged**